[Telegram] panic: runtime error: slice bounds out of range

**Describe the bug**
Matterbridge crashes with the following error:

```
panic: runtime error: slice bounds out of range

goroutine 386 [running]:
github.com/42wim/matterbridge/bridge/telegram.(*Btelegram).handleEntities(0xc000458420, 0xc0004d1e20, 0xc000628780)
        /go/src/github.com/42wim/matterbridge/bridge/telegram/handlers.go:378 +0x3b4
github.com/42wim/matterbridge/bridge/telegram.(*Btelegram).handleRecv(0xc000458420, 0xc0007f6cc0)
        /go/src/github.com/42wim/matterbridge/bridge/telegram/handlers.go:153 +0x43c
created by github.com/42wim/matterbridge/bridge/telegram.(*Btelegram).Connect
        /go/src/github.com/42wim/matterbridge/bridge/telegram/telegram.go:46 +0x376
```

**To Reproduce**
I don't know how to reproduce it. It happens once in a week.

**Environment (please complete the following information):**
- Linux
- version: 1.15.0 f06e9b5

--- a/bridge/telegram/handlers.go
+++ b/bridge/telegram/handlers.go
@@ -5,10 +5,11 @@ import (
 	"regexp"
 	"strconv"
 	"strings"
+	"unicode/utf16"
 
 	"github.com/42wim/matterbridge/bridge/config"
 	"github.com/42wim/matterbridge/bridge/helper"
-	"github.com/go-telegram-bot-api/telegram-bot-api"
+	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
 )
 
 func (b *Btelegram) handleUpdate(rmsg *config.Message, message, posted, edited *tgbotapi.Message) *tgbotapi.Message {
@@ -375,8 +376,13 @@ func (b *Btelegram) handleEntities(rmsg *config.Message, message *tgbotapi.Messa
 				b.Log.Errorf("entity text_link url parse failed: %s", err)
 				continue
 			}
-			link := rmsg.Text[e.Offset : e.Offset+e.Length]
-			rmsg.Text = strings.Replace(rmsg.Text, link, url.String(), 1)
+			utfEncodedString := utf16.Encode([]rune(rmsg.Text))
+			if e.Offset+e.Length > len(utfEncodedString) {
+				b.Log.Errorf("entity length is too long %d > %d", e.Offset+e.Length, len(utfEncodedString))
+				continue
+			}
+			link := utf16.Decode(utfEncodedString[e.Offset : e.Offset+e.Length])
+			rmsg.Text = strings.Replace(rmsg.Text, string(link), url.String(), 1)
 		}
 	}
 }

